src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -517,10 +575,61 @@ public int MultiDimensionalArrayRank
             }
         }
 
+        public bool IsValueType
+        {
+            get
+            {
+                return (Flags & enum_flag_Category_ValueType_Mask) == enum_flag_Category_ValueType;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public TypeHandle GetArrayElementTypeHandle()
+        {
+            return *(TypeHandle*)&((MethodTable*)Unsafe.AsPointer(ref this))->ElementType;
+        }
+
         [MethodImpl(MethodImplOptions.InternalCall)]
         public extern uint GetNumInstanceFieldBytes();
     }
 
+    /// <summary>
+    /// A type handle, which can wrap either a pointer to a <c>TypeDesc</c> or to a <see cref="MethodTable"/>.
+    /// </summary>
+    [StructLayout(LayoutKind.Explicit)]
```

Comment:
This does not need to be Explicit layout.

Reply:
Right. Fixed in 23b4a3ffa26e331db50ddce66353e82ce7913722.